crypto/engine/eng_devcrypto.c

```diff
@@ -664,7 +666,7 @@ void engine_load_devcrypto_int()
 # endif
 #endif
         || !ENGINE_set_ciphers(e, devcrypto_ciphers)
-#if defined(COP_FLAG_UPDATE) && defined(COP_FLAG_FINAL)
+#ifdef IMPLEMENT_DIGEST
         || !ENGINE_set_digests(e, devcrypto_digests)
 #endif
         ) {
```

Comment:
the error handling here is tricky, because if and only if the ENGINE_set_destroy_function(e, devcrypto_unload)
succeeded, the cfd will not leak

Reply:
maybe move that up to the first if, close cfd only there?